### DISCUSSION TITLE
Installing Hive on M1 macOs

### DISCUSSION THREAD

**Discussion by SainathBysani** [Homebrew/discussions#2723] [Getting started] [answered]
Hello there..
When I'm trying to install Hive through homebrew on Apple Silicon macOS it is throwing me below error 
"Error: openjdk@8: no bottle available!"

Command used: brew install hive

Same case is happening for Sqoop as well. 

Anyone has a solution for this?
Much appreciated in advance.

**Comment by SMillerDev** [comment, marked as answer]
You can install Homebrew with Rosetta, performance will be worse but OpenJDK 8 does not work with ARM processors

**Reply by SainathBysani** [reply]
Thanks for your reply. So, am I able to install Hive and Sqoop if I install homebrew with Rosetta?

**Reply by SMillerDev** [reply]
That I don't know, but the openjdk@8 problem you posted can only be resolved that way.

**Reply by SainathBysani** [reply]
Thanks for your response again. I did install Homebrew with Rosetta and below error I got
Error: Cannot install under Rosetta 2 in ARM default prefix (/opt/homebrew)!
To rerun under ARM use:
    arch -arm64 brew install ...
To install under x86_64, install Homebrew into /usr/local.

Can you please help me on how to install homebrew into /usr/local ? like any command that I can paste?

Thank you again for your support on this.

**Reply by gromgit** [reply]
I'm still waiting on my M1 Mac, but you should just need to run the official Homebrew installer with `arch -x86_64`, like this:
```
arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```
That should install a separate Homebrew hierarchy under `/usr/local`, that only does Intel bottles. Once it's completed successfully, installing Hive is simply:
```
/usr/local/bin/brew install hive
```
Note that you'll now have two separate Homebrew hierarchies on your machine, an Intel-only one under `/usr/local`, and an M1-only one under `/opt/homebrew`, so I suggest adding `/usr/local/bin` _after_ `/opt/homebrew/bin` in your PATH.

**Reply by SainathBysani** [reply]
@gromgit  - Man, you're awesome. I just tried the command that you have provided and BAM... It worked :) 
Basically I have duplicated a terminal and customized it to open only with Rosetta. I did execute the command in this terminal and am able to install both Hive and Sqoop successfully. 

Thank you once again for your support. Very much appreciated. 
Cheers!

**Reply by zhiqiangsdevo** [reply]
> 

Hello gromgit,
I came across this solution. What do you mean by adding /usr/local/bin after /opt/homebrew/bin in the PATH. How to do it?

**Reply by gromgit** [reply]
Actually, if I remember correctly, macOS adds `/usr/local/bin` to the PATH by default, before `brew shellenv` adds `/opt/homebrew/bin` in front, so all you need to do is have `eval "$(/opt/homebrew/bin/brew shellenv)"` in your shell startup file.

So if you followed the Homebrew installer's _Next steps_ instructions correctly, there's nothing else you need to do.

**Reply by abhi413sh** [reply]
No available formula with the name "hive". 
i am getting the following error
do helpme with it!

**Reply by abhi413sh** [reply]
if its posiible please give all the commands needed to run

**Reply by gromgit** [reply]
Have you done what I mentioned here? https://github.com/Homebrew/discussions/discussions/2723#discussioncomment-1906993

If you have, and you still got the error you mentioned, post the output of the following commands:
```
/usr/local/bin/brew config
/usr/local/bin/brew doctor
echo $PATH
```

**Reply by abhi413sh** [reply]
i am getting these outputs

**Reply by SMillerDev** [reply]
```
Homebrew/homebrew-core was not tapped properly! Run:
rm -rf "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core"
brew tap homebrew/core
```

Try following this advice

**Reply by abhi413sh** [reply]
ok i have done this

**Reply by abhi413sh** [reply]
HiveException java.lang.RuntimeException: Unable to instantiate org.apache.hadoop.hive.ql.metadata.SessionHiveMetaStoreClient

**Reply by Mourya1319** [reply]
Hey @SMillerDev , I have followed the steps suggested by you. I am getting the error for  /usr/local/bin/brew install hive

Error: Cannot install in Homebrew on ARM processor in Intel default prefix (/usr/local)!
Please create a new installation in /opt/homebrew using one of the
"Alternative Installs" from:
  https://docs.brew.sh/Installation
You can migrate your previously installed formula list with:
  brew bundle dump

The outputs for the 3 commands you asked for are below:
1. HOMEBREW_VERSION: 4.1.22
ORIGIN: https://github.com/Homebrew/brew
HEAD: 251a098fbf1c02ca95f53b4806e5b6da8d5e4a80
Last commit: 2 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: c5b51fdd139699da601a23549cd7e7a2a3f8f139
Core tap last commit: 46 minutes ago
Core tap branch: master
Core tap JSON: 27 Nov 17:35 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.88.1 => /usr/bin/curl
macOS: 13.4-arm64
CLT: 14.3.1.0.1.1683849156
Xcode: N/A
Rosetta 2: false

2. Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You have an unnecessary local Core tap!
This can cause problems installing up-to-date formulae.
Please remove it by running:
 brew untap homebrew/core

Warning: Your Homebrew's prefix is not /opt/homebrew.

Many of Homebrew's bottles (binary packages) can only be used with the default prefix.
Consider uninstalling Homebrew and reinstalling into the default prefix.
It is expected behaviour that some formulae will fail to build in this unsupported configuration.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.


3./bin/:/bin/:/bin/:/Library/Frameworks/Python.framework/Versions/3.11/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Library/Java/JavaVirtualMachines/jdk-20.jdk/Contents/Home/bin:/opt/homebrew/Cellar/hadoop/3.3.6/libexec/bin:/opt/homebrew/Cellar/hadoop/3.3.6/libexec/sbin:/Library/Java/JavaVirtualMachines/jdk-20.jdk/Contents/Home/bin:/opt/homebrew/Cellar/hadoop/3.3.6/libexec/bin:/opt/homebrew/Cellar/hadoop/3.3.6/libexec/sbin

Can you please help me with this?

**Reply by ZhongRuoyu** [reply]
@Mourya1319 Please try `arch -x86_64 /usr/local/bin/brew install hive` instead.

**Reply by Mourya1319** [reply]
@ZhongRuoyu , thanks for your reply and support. I will try this command and keep you posted.

**Comment by alberttwong** [comment]
I think if we switch to temurin@8, it'll work (assumption that rosetta has been installed on your mac silicon).   I don't have the knowledge to do this.
